src/client/components/configuration/tree/tree.tsx

```diff
@@ -0,0 +1,66 @@
+import * as classnames from 'classnames'
+import { observer } from 'mobx-react'
+import * as React from 'react'
+
+import FileIcon from './file.svg'
+import FolderIconOpen from './folder-open.svg'
+import FolderIcon from './folder.svg'
+import * as style from './style.css'
+
+export interface Node {
+  label: string
+  expanded: boolean
+  leaf: boolean
+  selected: boolean
+  children?: Node[]
+}
+
+export interface TreeProps {
+  data: Node
+  level?: number
+  onClick(node: Node): void
+}
+
+@observer
+export class Tree extends React.Component<TreeProps, any> {
+  constructor(props: any, context: any) {
+    super(props, context)
+    this.onClick = this.onClick.bind(this)
+  }
+
+  public render(): JSX.Element {
+    const children = this.props.data.children || []
+    const hasChildren = children.length > 0
+    const level = this.props.level || 0
+    const classes = classnames(
+      style.treenode,
+      { [style['treenode--expanded']]: this.props.data.expanded },
+      { [style['treenode--selected']]: this.props.data.selected },
+    )
+    const headerInlineStyle = { paddingLeft: 8 + (level * 22) + 'px' }
+
+    return (
+      <div className={classes}>
+        <div className={style.treenode__header} onClick={this.onClick} style={headerInlineStyle}>
```

Comment:
Could we use `ul` and `li` for this HTML structure so we don't need to use a calculated `paddingLeft`?

http://jsbin.com/zoyabovami/edit?html,css,output

Credit to @MonicaOlejniczak for the idea 🙂